DON-16 JWT 인증 기능 구현

ksw6823 opened the pull request:
- JWT 기반 인증 시스템 구현 (passport-jwt 사용)
- AuthController, AuthService, JwtStrategy 추가
- 로그인/토큰재발급/로그아웃 API 구현
- JWT 관련 DTO 및 Guard 구현
- 역할 기반 접근 제어 시스템 추가
- 관련 의존성 패키지 설치

ksw6823 committed:
DON-16 JWT 인증 기능 구현

- JWT 기반 인증 시스템 구현 (passport-jwt 사용)
- AuthController, AuthService, JwtStrategy 추가
- 로그인/토큰재발급/로그아웃 API 구현
- JWT 관련 DTO 및 Guard 구현
- 역할 기반 접근 제어 시스템 추가
- 관련 의존성 패키지 설치

ksw6823 committed:
Merge branch 'develop' into feature/don-16_JWT

lightpurple reviewed (changes requested):
indent를 space 4 통일 후 다시 pr 주세요!

ksw6823 committed:
fix: indent space 4로 변경

lightpurple commented on src/v1/auth/auth.module.ts:
```diff
@@ -0,0 +1,33 @@
+            useFactory: async (configService: ConfigService) => ({
+                secret:
+                    configService.get<string>('JWT_ACCESS_SECRET') ||
+                    'default-secret',
```
JWT_ACCESS_SECRET이 없을때 'default-secret'을 사용하는건 보안상 굉장히 안좋은 영향을 끼칠것 같아요! 보안관련해서 환경변수가 없을땐 에러처리하는게 맞습니다.

lightpurple commented on src/v1/users/users.service.ts:
```diff
@@ -56,4 +64,44 @@ export class UsersService {
+        const user = await this.userRepository.findOne({ 
+            where: { login_id }
+        });
+        return user || null;
```
Return값이 user or null인 이유가 있나요?

lightpurple commented on src/v1/users/users.service.ts:
```diff
@@ -56,4 +64,44 @@ export class UsersService {
+    // 로그인 아이디로 사용자 조회 (JWT 인증용)
+    // login_id: 로그인 아이디
+    // return: 사용자 또는 null
+    async findByLoginId(login_id: string): Promise<User | null> {
```
findOne으로 통합 가능할 거 같은데 따로 분리한 이유가 있나요?

lightpurple commented on src/v1/auth/auth.service.ts:
```diff
@@ -0,0 +1,134 @@
+        const unit = expirationTime.slice(-1);
+        const value = parseInt(expirationTime.slice(0, -1), 10);
+
+        switch (unit) {
```
lowerCase 해주세요

lightpurple commented on src/v1/auth/auth.service.ts:
```diff
@@ -0,0 +1,134 @@
+        
+        // 액세스 토큰 생성
+        const accessToken = this.jwtService.sign(payload, {
+            secret: this.configService.get<string>('JWT_ACCESS_SECRET') || 'default-secret',
```
위 코멘트와 동일

lightpurple commented on src/v1/users/users.service.ts:
```diff
@@ -56,4 +64,44 @@ export class UsersService {
+    // 사용자 리프레시 토큰 업데이트 (JWT 인증용)
+    // userId: 사용자 ID
+    // refreshToken: 리프레시 토큰
+    async updateRefreshToken(userId: number, refreshToken: string): Promise<void> {
```
이것도 update와 분리한 이유가 있나요?

lightpurple replied:
수정하라는게 아니라 auth.service.ts에서 usersService의 update를 안쓰고 updateRefreshToken 메서드를 따로 만들어서 이유가 궁금했던건데..
일단은 머지할게요

ksw6823 committed:
fix: JWT 인증 관련 리뷰 피드백 반영

- JWT 환경변수 기본값 제거 및 보안 강화
- 환경변수명을 소문자로 변경 (jwt_access_secret 등)
- UsersService 메서드 통합 및 코드 개선
- 리뷰어 피드백: 보안 강화, lowerCase 적용
